lib/fs.js

```diff
@@ -1329,14 +1330,32 @@ fs.chown = function(path, uid, gid, callback) {
   if (handleError((path = getPathFromURL(path)), callback))
     return;
   if (!nullCheck(path, callback)) return;
-  var req = new FSReqWrap();
+
+  if (typeof path !== 'string' && !isUint8Array(path)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'path',
+                               ['string', 'Buffer', 'URL']);
+  }
+  if (!Number.isInteger(uid))
```

Comment:
Should we also check for `uid >= 0`? Same question for the other checks in this commit

Reply:
I do believe that uid and gid both can be negative in some edge cases.